Реализованы экран регистрации, экран с кошельком и экраны создания транзакции.

vk59 opened the pull request.

Commit:
Реализован Navigation Component. Дополнен .gitignore.

Commit:
Реализован splash-screen.

gera2705 committed:
Добавлена разметка для OnBoardingFragment.kt

gera2705 committed:
Merge branch 'feature/splash' of https://github.com/vk59/KoshelOK into feature/onboarding

 Conflicts:
	.idea/vcs.xml
	app/build.gradle
	app/src/main/java/com/tinkoff_sirius/koshelok/ui/OnBoardingFragment.kt
	app/src/main/java/com/tinkoff_sirius/koshelok/ui/OnBoardingViewModel.kt
	app/src/main/res/layout/on_boarding_fragment.xml

Commit:
Реализована авторизация.

gera2705 committed:
Коммит перед разработкой адаптера.

gera2705 committed:
Merge branch 'feature/auth' of https://github.com/vk59/KoshelOK into feature/main

Commit:
Code refactor.

Commit:
Перемещена логика сохранения данных аккаунта в AccountShared.kt.

Commit:
Изменена обработка nullable значений в getPreferences(). Добавлены отступы. Подключен Timber.

vk59 committed:
Merge pull request #1 from vk59/feature/auth

Реализована авторизация.

gera2705 committed:
Сделан ресайкл вью, на главном экране

gera2705 committed:
Убрал TODO.

Commit:
Merge remote-tracking branch 'origin/feature/main' into feature/auth.

vk59 committed:
Merge pull request #2 from vk59/feature/auth

Feature/auth

gera2705 committed:
Merge branch 'develop' of https://github.com/vk59/KoshelOK into feature/main

Commit:
Реализован главный экран.

gera2705 committed:
Добавлен экран fragment_operation_type.xml и fragment_set_sum.xml, дополнен nav_graph

Commit:
Рефактор кода. Добавлен класс AppConfig.kt для хранения локальных данных. Дополнена основная тема приложения.

Commit:
Добавлен TransactionType.kt.

gera2705 committed:
Добавлен экран transaction_category_fragment

Commit:
Merge remote-tracking branch 'origin/feature/create_transaction' into merge

# Conflicts:
#	.idea/misc.xml
#	app/src/main/java/com/tinkoff_sirius/koshelok/adapters/DateViewHolder.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/model/MainAdapterItem.kt
#	app/src/main/res/navigation/nav_graph.xml
#	app/src/main/res/values/colors.xml
#	app/src/main/res/values/strings.xml
#	app/src/main/res/values/themes.xml

Commit:
Объединены главный экран с созданием транзакции

Commit:
Реализованы транзакции на главном экране

Commit:
Транзакции дополнены актуальной датой и временем.

Commit:
Добавлен Detekt. Рефактор кода.

gera2705 committed:
Реализован экран с редактированием транзакции

gera2705 committed:
Реализован экран с редактированием транзакции

gera2705 committed:
Merge branch 'feature/main_full' of https://github.com/vk59/KoshelOK into feature/create_transaction

 Conflicts:
	.idea/misc.xml
	app/src/main/java/com/tinkoff_sirius/koshelok/ui/operation/OperationTypeFragment.kt
	app/src/main/java/com/tinkoff_sirius/koshelok/ui/transaction_category/adapters/TransactionCategoryAdapter.kt

Commit:
Реализовано удаление и редактирование транзакции.

Commit:
Изменен класс с категориями.

Commit:
Добавлена фабрика по созданию EncryptedSharedPreferences.

Commit:
Добавлена фабрика по созданию EncryptedSharedPreferences.

Commit:
Добавлена фабрика по созданию EncryptedSharedPreferences.

gera2705 committed:
До внедрения SharedPref

gera2705 committed:
До внедрения SharedPref

gera2705 committed:
Merge branch 'feature/main_full' of https://github.com/vk59/KoshelOK into feature/create_transaction

Commit:
Рефактор кода.

Commit:
RecyclerView by lazy изменен на инциализацию внутри onViewCreated.

lexa-diky commented on .idea/jarRepositories.xml:
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
.idea - в gitignore

lexa-diky commented on app/build.gradle:
```diff
@@ -30,16 +31,57 @@ android {
+    // Navigation
+    implementation 'androidx.navigation:navigation-fragment-ktx:2.3.5'
+    implementation 'androidx.navigation:navigation-ui-ktx:2.3.5'
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
```
Зачем?

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/model/Category.kt:
```diff
@@ -0,0 +1,12 @@
+package com.tinkoff_sirius.koshelok.model
+
+sealed class Category(open val type: String, open val icon: Int, open val color: Int) {
```
Кажется делать sealed лишнее, или я не совсем понимаю, зачем

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/model/CategorySealed.kt:
```diff
@@ -0,0 +1,70 @@
+                get() = R.drawable.ic_present
+        }
+
+        class Sport(override val color: Int = R.color.green) : Outcome() {
```
Это все можно обычным enum сделать

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/model/Transaction.kt:
```diff
@@ -0,0 +1,9 @@
+import kotlinx.datetime.LocalDate
+
+data class Transaction(
+    val sum: Double,
```
BigDecimal

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/model/Types.kt:
```diff
@@ -0,0 +1,12 @@
+package com.tinkoff_sirius.koshelok.model
+
+enum class Types(val nameType: String) {
+    SALARY("Зарплата"),
```
В локализацию

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/repository/AccountShared.kt:
```diff
@@ -0,0 +1,21 @@
+import android.content.SharedPreferences
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount
+
+class AccountShared(private val sharedPreferences: SharedPreferences) {
```
Account -> AccountRepository

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/MainFragment.kt:
```diff
@@ -0,0 +1,112 @@
+class MainFragment : Fragment() {
+
+    private val mainRecyclerAdapter by lazy {
+        MainRecyclerAdapter(object : OptionsCallback {
```
Уберите разнесите callback на 2 лямбды

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/MainFragment.kt:
```diff
@@ -0,0 +1,112 @@
+
+    private val viewModel: MainViewModel by viewModels()
+    private val binding by viewBinding(FragmentMainBinding::bind)
+    private var exitFlag = false
```
Стейт в Fragment хранить нельзя

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/MainFragment.kt:
```diff
@@ -0,0 +1,112 @@
+    private val binding by viewBinding(FragmentMainBinding::bind)
+    private var exitFlag = false
+
+    private lateinit var recyclerView: RecyclerView
```
ViewBinding

lexa-diky replied:
Его хранить отдельно не нужно, каждый раз из биндинга доставайте

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/MainViewModel.kt:
```diff
@@ -0,0 +1,74 @@
+    }
+
+    private fun createNewMainItemList() {
+        val transItems = mutableListOf<MainItem>()
```
listOf(item1, *otherItems.toTypedArray()) или buildList. Второй вариант - лучше

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/MainViewModel.kt:
```diff
@@ -0,0 +1,74 @@
+    private fun createNewMainItemList() {
+        val transItems = mutableListOf<MainItem>()
+        val transDate = transactions.groupBy { it.date }
+        val currentTime = Clock.System.now().toLocalDateTime(TimeZone.currentSystemDefault())
```
.toLocalDateTime(TimeZone.currentSystemDefault()) - вот с такими штуками в бизнесс логике - осторожно, лучше все считать в +0 а потом в UI форматировать в локальное время

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/MainViewModel.kt:
```diff
@@ -0,0 +1,74 @@
+            val dateText =
+                when (currentDate.minus(date.toLocalDate())) {
+                    DatePeriod(0, 0, 0) -> "Сегодня"
+                    DatePeriod(0, 0, 1) -> "Вчера"
```
В локализацию

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/MainViewModel.kt:
```diff
@@ -0,0 +1,74 @@
+
+        transactionsItems = transItems.toList()
+
+        val header = AppConfig.headerExample.toMutableList()
```
Чую баги. Создаете новый mutable лист и что-то в него добавляете, выглядит странно.

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/MainViewModel.kt:
```diff
@@ -0,0 +1,74 @@
+    }
+
+    private fun toString(localDate: LocalDate): String {
+        val result = StringBuffer().append("${localDate.dayOfMonth} ")
```
StringBuffer -> StringBuilder

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/MainViewModel.kt:
```diff
@@ -0,0 +1,74 @@
+        val result = StringBuffer().append("${localDate.dayOfMonth} ")
+        result.append(
+            when (localDate.monthNumber) {
+                1 -> "января"
```
В локализацию

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/adapters/MainRecyclerAdapter.kt:
```diff
@@ -0,0 +1,56 @@
+class MainRecyclerAdapter(private val callback: OptionsCallback) :
+    RecyclerView.Adapter<MainViewHolder>() {
+
+    fun setData(data: List<MainItem>) {
```
public val  -> private val -> public fun  -> private fun

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/adapters/MainRecyclerAdapter.kt:
```diff
@@ -0,0 +1,56 @@
+        return when (viewType) {
+            TYPE_TRANSACTION -> TransactionViewHolder(
+                inflater.inflate(R.layout.item_transaction, parent, false),
+            callback
```
Не понял, что это

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/adapters/MainRecyclerAdapter.kt:
```diff
@@ -0,0 +1,56 @@
+            )
+            TYPE_HEADER -> HeaderViewHolder(inflater.inflate(R.layout.item_home_header, parent, false))
+            TYPE_DATE -> DateViewHolder(inflater.inflate(R.layout.item_date, parent, false))
+            else -> @Suppress throw Exception("You should use Transaction and Header View Holder")
```
@Suppress - зачем?

lexa-diky replied:
error("You should use Transaction and Header View Holder")

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/adapters/TransactionsDiffUtils.kt:
```diff
@@ -0,0 +1,14 @@
+import com.tinkoff_sirius.koshelok.ui.main.adapters.model.MainItem
+
+class TransactionsDiffUtils : DiffUtil.ItemCallback<MainItem>() {
+    override fun areItemsTheSame(oldItem: MainItem, newItem: MainItem): Boolean =
```
Строку сверху

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/adapters/model/Month.kt:
```diff
@@ -0,0 +1,16 @@
+package com.tinkoff_sirius.koshelok.ui.main.adapters.model
+
+enum class Month(val number: Int, val monthName: String) {
```
В либе уже есть

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/adapters/MainRecyclerAdapter.kt:
```diff
@@ -0,0 +1,56 @@
+    private val diff = AsyncListDiffer(this, TransactionsDiffUtils())
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MainViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
```
ViewBinding умеет инфлейтить, юзайте его

lexa-diky replied:
override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            ApplicationFormsItemBottomsheetListBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )
        )

lexa-diky replied:
class ViewHolder(val viewBinding: ApplicationFormsItemBottomsheetListBinding) : RecyclerView.ViewHolder(viewBinding.root)

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/adapters/view_holders/HeaderViewHolder.kt:
```diff
@@ -0,0 +1,59 @@
+
+    private val binding: ItemHomeHeaderBinding by viewBinding(ItemHomeHeaderBinding::bind)
+
+    private val walletCurrentMoney: TextView by lazy {
```
Сразу из binding доставайте

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/adapters/view_holders/HeaderViewHolder.kt:
```diff
@@ -0,0 +1,59 @@
+            cardIncome.findViewById<TextView>(R.id.typeCard).text =
+                view.context.resources.getString(R.string.card_income_title)
+
+            cardIncome.findViewById<ImageView>(R.id.icon).setImageResource(R.drawable.ic_green_point)
```
custom view + binding

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/adapters/view_holders/TransactionViewHolder.kt:
```diff
@@ -0,0 +1,42 @@
+    override fun bind(data: MainItem) {
+        if (data is MainItem.Transaction) {
+            val type = when (data.category) {
+                is Category.Income -> "Пополнение"
```
локализация

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/adapters/view_holders/TransactionViewHolder.kt:
```diff
@@ -0,0 +1,42 @@
+import com.tinkoff_sirius.koshelok.ui.main.adapters.model.MainItem
+
+class TransactionViewHolder(private val itemView: View, private val callback: OptionsCallback)
+    : MainViewHolder(itemView) {
```
ditto

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/adapters/view_holders/TransactionViewHolder.kt:
```diff
@@ -0,0 +1,42 @@
+                categoryTransaction.text = data.category.type
+                moneyTransaction.text =
+                    if (type == "Траты") "-${data.sum}" else data.sum.toString()
+                typeTransaction.text = type
```
Локализация

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/on_boarding/OnBoardingFragment.kt:
```diff
@@ -0,0 +1,78 @@
+    override fun onStart() {
+        super.onStart()
+        val account = GoogleSignIn.getLastSignedInAccount(requireContext())
+        Timber.tag("ACCOUNT").d("${account?.displayName}  ${account?.email}")
```
tag(...) - можно и без Timber умеет сам генерить, но это как хотите

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/on_boarding/OnBoardingFragment.kt:
```diff
@@ -0,0 +1,78 @@
+        super.onStart()
+        val account = GoogleSignIn.getLastSignedInAccount(requireContext())
+        Timber.tag("ACCOUNT").d("${account?.displayName}  ${account?.email}")
+        if (account != null) {
```
account?.let { navigateWith(it) }

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/on_boarding/OnBoardingFragment.kt:
```diff
@@ -0,0 +1,78 @@
+
+    private fun navigateWith(account: GoogleSignInAccount) {
+        AccountShared(SharedPreferencesFactory().create(requireContext())).saveAccount(account)
+        findNavController().navigate(R.id.action_onBoardingFragment_to_mainFragment)
```
findNavController() - lazy - долгая операция

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/operation/OperationTypeFragment.kt:
```diff
@@ -0,0 +1,42 @@
+
+    private fun initListeners(v: View) {
+        binding.setType.setOnClickListener {
+            v.findNavController()
```
v - что это?, не понятно

lexa-diky replied:
В фрагменте есть аксессор для view

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/set_sum/SetSumFragment.kt:
```diff
@@ -0,0 +1,60 @@
+class SetSumFragment : Fragment() {
+    private val binding: FragmentSetSumBinding by viewBinding(FragmentSetSumBinding::bind)
+
+    var pref: SharedPreferences? = null
```
Убюрать в repo

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/set_sum/SetSumFragment.kt:
```diff
@@ -0,0 +1,60 @@
+        super.onViewCreated(view, savedInstanceState)
+
+        pref = this.activity?.getSharedPreferences("CREATED_TRANSACTION", Context.MODE_PRIVATE)
+        binding.sumText.setText(pref?.getString("sum", ""))
```
ditto

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/set_sum/SetSumFragment.kt:
```diff
@@ -0,0 +1,60 @@
+        binding.setSumButton.setOnClickListener {
+            if (!binding.sumText.text?.trim().isNullOrEmpty() && !binding.sumText.text?.toString().equals(".")) {
+                saveData(binding.sumText.text.toString())
+                v.findNavController()
```
ditto

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/set_sum/SetSumFragment.kt:
```diff
@@ -0,0 +1,60 @@
+
+    private fun initListeners(v: View) {
+        binding.setSumButton.setOnClickListener {
+            if (!binding.sumText.text?.trim().isNullOrEmpty() && !binding.sumText.text?.toString().equals(".")) {
```
разнести по переменным

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/set_sum/SetSumFragment.kt:
```diff
@@ -0,0 +1,60 @@
+                v.findNavController()
+                    .navigate(R.id.action_setSumFragment_to_operationTypeFragment)
+            } else {
+                Toast.makeText(requireContext(), "Введите сумму!", Toast.LENGTH_LONG).show()
```
Локализация

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/set_sum/SetSumFragment.kt:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    fun saveData(sum: String){
+        val editor = pref?.edit()
```
в repo через viewmodel

lexa-diky commented on app/src/main/java/com/tinkoff_sirius/koshelok/ui/transaction_category/adapters/TransactionCategoryViewHolder.kt:
```diff
@@ -0,0 +1,24 @@
+
+    init {
+        mTransactionCategoryImage = itemView.findViewById(R.id.transaction_category_image)
+        mTransactionCategoryTitle = itemView.findViewById(R.id.transaction_category_title)
```
viewbinding

Commit:
Исправлен внешний вид кнопок.

Commit:
Рефактор кода в Main (fragment, viewModel, adapters, viewHolders).

gera2705 committed:
Реализована работа с sharedPref (не работает)

Commit:
Init RxJava3 for Main.

Commit:
Имитация NetworkService и репозитория для MainFragment.

Commit:
Имитация NetworkService и репозитория для MainFragment.

Имитация NetworkService и репозитория для MainFragment.

Имитация NetworkService и репозитория для MainFragment.

gera2705 committed:
Добавлено сохранения суммы и типа на экранах создания транзакции

gera2705 committed:
Перед pull из develop

Commit:
Внедрение зависимостей. Правки в MainViewModel, MainFragment.

Commit:
Добавлен ResourceProvider.kt.

gera2705 committed:
Реализовано создание транзакции

gera2705 committed:
Коммит после починки гита

gera2705 committed:
Коммит после починки гита 2

Commit:
Промежуточный коммит.

Commit:
Промежуточный коммит.

Commit:
Merge branch 'feature/create_transaction' into merge

# Conflicts:
#	.idea/misc.xml
#	app/src/main/java/com/tinkoff_sirius/koshelok/KoshelokApp.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/config/AppConfig.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/entities/Category.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/entities/Transaction.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/entities/Types.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/repository/SharedPreferencesFactory.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/repository/WalletRepository.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/ui/MainActivity.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/MainFragment.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/MainViewModel.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/OptionsCallback.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/adapters/MainRecyclerAdapter.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/adapters/view_holders/DateViewHolder.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/adapters/view_holders/HeaderViewHolder.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/ui/main/adapters/view_holders/TransactionViewHolder.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/ui/operation/OperationTypeFragment.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/ui/set_sum/SetSumFragment.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/ui/transaction_category/TransactionCategoryFragment.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/ui/transaction_category/adapters/TransactionCategoryAdapter.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/ui/transaction_category/adapters/TransactionCategoryViewHolder.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/ui/transaction_editing/TransactionEditingFragment.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/ui/transaction_editing/TransactionEditingViewModel.kt
#	app/src/main/java/com/tinkoff_sirius/koshelok/ui/transaction_editing/TransactionItemEditingView.kt
#	app/src/main/res/layout/fragment_operation_type.xml
#	app/src/main/res/layout/fragment_transaction_editing.xml
#	app/src/main/res/layout/item_transaction_editing.xml
#	app/src/main/res/navigation/nav_graph.xml
#	app/src/main/res/values/attrs.xml

Commit:
Merge commit.

Commit:
Package refactor.

gera2705 committed:
Добавлена отчистка sharedPref файла с транзакцией

gera2705 committed:
Проведена чистка кода

Commit:
Дополнен файл Api.kt

Commit:
Добавлены импорты. Добавлена иконка.

gera2705 committed:
Merge remote-tracking branch 'origin/develop' into develop

Commit:
Внедрена зависимость viewModelFactory в экраны создания транзакций.

Commit:
Добавлено сохранение транзакций на моковый сервер.

Commit:
Дополнена CategoryData.kt (структура как в Category.kt).
